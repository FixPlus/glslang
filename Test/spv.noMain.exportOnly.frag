#version 460 core

float foo(float, float);

void bar(inout vec3 v, int n)
{
    v.x += n;
}

int bar(vec3 v)
{
    return int(v.x + v.y + v.z);
}

float foo(int n, float s)
{
    return s * float(n);
}

float foo(float a, float b)
{
    return a + b;
}
