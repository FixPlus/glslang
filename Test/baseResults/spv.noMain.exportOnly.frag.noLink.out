spv.noMain.exportOnly.frag
Shader version: 460
gl_FragCoord origin is upper left
0:? Sequence
0:5  Function Definition: bar(vf3;i1; ( global void)
0:5    Function Parameters: 
0:5      'v' ( inout highp 3-component vector of float)
0:5      'n' ( in highp int)
0:7    Sequence
0:7      add second child into first child ( temp highp float)
0:7        direct index ( temp highp float)
0:7          'v' ( inout highp 3-component vector of float)
0:7          Constant:
0:7            0 (const int)
0:7        Convert int to float ( temp highp float)
0:7          'n' ( in highp int)
0:10  Function Definition: bar(vf3; ( global highp int)
0:10    Function Parameters: 
0:10      'v' ( in highp 3-component vector of float)
0:12    Sequence
0:12      Branch: Return with expression
0:12        Convert float to int ( temp int)
0:12          add ( temp highp float)
0:12            add ( temp highp float)
0:12              direct index ( temp highp float)
0:12                'v' ( in highp 3-component vector of float)
0:12                Constant:
0:12                  0 (const int)
0:12              direct index ( temp highp float)
0:12                'v' ( in highp 3-component vector of float)
0:12                Constant:
0:12                  1 (const int)
0:12            direct index ( temp highp float)
0:12              'v' ( in highp 3-component vector of float)
0:12              Constant:
0:12                2 (const int)
0:15  Function Definition: foo(i1;f1; ( global highp float)
0:15    Function Parameters: 
0:15      'n' ( in highp int)
0:15      's' ( in highp float)
0:17    Sequence
0:17      Branch: Return with expression
0:17        component-wise multiply ( temp highp float)
0:17          's' ( in highp float)
0:17          Convert int to float ( temp highp float)
0:17            'n' ( in highp int)
0:20  Function Definition: foo(f1;f1; ( global highp float)
0:20    Function Parameters: 
0:20      'a' ( in highp float)
0:20      'b' ( in highp float)
0:22    Sequence
0:22      Branch: Return with expression
0:22        add ( temp highp float)
0:22          'a' ( in highp float)
0:22          'b' ( in highp float)
0:?   Linker Objects
0:?     Linker Function: foo(f1;f1; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp float)
0:?         '' ( in highp float)
0:?     Linker Function: bar(vf3;i1; ( global void)
0:?       Function Parameters: 
0:?         '' ( inout highp 3-component vector of float)
0:?         '' ( in highp int)
0:?     Linker Function: bar(vf3; ( global highp int)
0:?       Function Parameters: 
0:?         '' ( in highp 3-component vector of float)
0:?     Linker Function: foo(i1;f1; ( global highp float)
0:?       Function Parameters: 
0:?         '' ( in highp int)
0:?         '' ( in highp float)

Validation failed
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 60

                              Capability Shader
                              Capability Linkage
               1:             ExtInstImport  "GLSL.std.450"
                              MemoryModel Logical GLSL450
                              Source GLSL 460
                              Name 11  "bar(vf3;i1;"
                              Name 9  "v"
                              Name 10  "n"
                              Name 15  "bar(vf3;"
                              Name 14  "v"
                              Name 21  "foo(i1;f1;"
                              Name 19  "n"
                              Name 20  "s"
                              Name 26  "foo(f1;f1;"
                              Name 24  "a"
                              Name 25  "b"
                              Decorate 26(foo(f1;f1;) Linkage Attributes 678391654 1715155302 15153 0
                              Decorate 11(bar(vf3;i1;) Linkage Attributes 678584674 993224310 3879273 0
                              Decorate 15(bar(vf3;) Linkage Attributes 678584674 993224310 0 0
                              Decorate 21(foo(i1;f1;) Linkage Attributes 678391654 1715155305 15153 0
               2:             TypeFloat 32
               3:             TypeVector 2(float) 3
               4:             TypePointer Function 3(fvec3)
               5:             TypeInt 32 1
               6:             TypePointer Function 5(int)
               7:             TypeVoid
               8:             TypeFunction 7 4(ptr) 6(ptr)
              13:             TypeFunction 5(int) 4(ptr)
              17:             TypePointer Function 2(float)
              18:             TypeFunction 2(float) 6(ptr) 17(ptr)
              23:             TypeFunction 2(float) 17(ptr) 17(ptr)
              30:             TypeInt 32 0
              31:     30(int) Constant 0
              38:     30(int) Constant 1
              42:     30(int) Constant 2
 11(bar(vf3;i1;):           7 Function None 8
            9(v):      4(ptr) FunctionParameter
           10(n):      6(ptr) FunctionParameter
              12:             Label
              28:      5(int) Load 10(n)
              29:    2(float) ConvertSToF 28
              32:     17(ptr) AccessChain 9(v) 31
              33:    2(float) Load 32
              34:    2(float) FAdd 33 29
              35:     17(ptr) AccessChain 9(v) 31
                              Store 35 34
                              Return
                              FunctionEnd
    15(bar(vf3;):      5(int) Function None 13
           14(v):      4(ptr) FunctionParameter
              16:             Label
              36:     17(ptr) AccessChain 14(v) 31
              37:    2(float) Load 36
              39:     17(ptr) AccessChain 14(v) 38
              40:    2(float) Load 39
              41:    2(float) FAdd 37 40
              43:     17(ptr) AccessChain 14(v) 42
              44:    2(float) Load 43
              45:    2(float) FAdd 41 44
              46:      5(int) ConvertFToS 45
                              ReturnValue 46
                              FunctionEnd
  21(foo(i1;f1;):    2(float) Function None 18
           19(n):      6(ptr) FunctionParameter
           20(s):     17(ptr) FunctionParameter
              22:             Label
              49:    2(float) Load 20(s)
              50:      5(int) Load 19(n)
              51:    2(float) ConvertSToF 50
              52:    2(float) FMul 49 51
                              ReturnValue 52
                              FunctionEnd
  26(foo(f1;f1;):    2(float) Function None 23
           24(a):     17(ptr) FunctionParameter
           25(b):     17(ptr) FunctionParameter
              27:             Label
              55:    2(float) Load 24(a)
              56:    2(float) Load 25(b)
              57:    2(float) FAdd 55 56
                              ReturnValue 57
                              FunctionEnd
